api/handlers/refererhandler.py

```diff
@@ -88,6 +96,31 @@ def post(self, cont_name, cid):
         else:
             self.abort(500, 'Analysis not added for container {} {}'.format(cont_name, cid))
 
+    def put(self, cont_name, **kwargs):
+        cid = kwargs.pop('cid')
+        _id = kwargs.pop('_id')
+
+        parent = self.storage.get_parent(cont_name, cid)
+        permchecker = self.get_permchecker(parent)
+        permchecker(noop)('PUT')
+
+
+        payload = self.request.json_body
```

Comment:
We do this everywhere, it would be nice to have a decorator such as `@verify_payload_exists` that just does this for us. Feel free to add one if you'd like :)